context/app/static/js/components/detailPage/ProcessedData/HelperPanel/HelperPanel.tsx

```diff
@@ -0,0 +1,251 @@
+        </HelperPanelBodyItem>
+      )}
+      {currentDataset.description && (
+        <HelperPanelBodyItem label="Description" noWrap>
+          {currentDataset.description}
+        </HelperPanelBodyItem>
+      )}
+      <HelperPanelBodyItem label="Pipeline">{currentDataset.pipeline}</HelperPanelBodyItem>
+      <HelperPanelBodyItem label="Consortium">{currentDataset.group_name}</HelperPanelBodyItem>
+      <HelperPanelBodyItem label={dateLabel}>{date && formatDate(date, 'yyyy-MM-dd')}</HelperPanelBodyItem>
+    </>
+  );
+}
+
+function WorkspaceButton() {
+  const currentDataset = useCurrentDataset();
+  const { isWorkspacesUser } = useAppContext();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const track = useTrackEntityPageEvent();
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    track({
+      action: 'Open Workspace Menu',
+      label: currentDataset?.hubmap_id,
+    });
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const {
+    control,
+    errors,
+    setDialogIsOpen: setOpenCreateWorkspace,
+    dialogIsOpen: createWorkspaceIsOpen,
+    ...rest
+  } = useCreateWorkspaceForm({ defaultName: currentDataset?.hubmap_id });
+
+  const openEditWorkspaceDialog = useOpenDialog('ADD_DATASETS_FROM_SEARCH');
+
+  if (!isWorkspacesUser || currentDataset?.status !== 'Published') {
+    return null;
+  }
+  // The selectable table provider is used here since a lot of the workspace logic relies on the selected rows
+  return (
+    <SelectableTableProvider tableLabel="Current Dataset" selectedRows={new Set([currentDataset.uuid])}>
+      <HelperPanelButton
+        startIcon={<WorkspacesIcon color="primary" />}
+        onClick={handleClick}
+        aria-controls={open ? 'basic-menu' : undefined}
+        aria-haspopup="true"
+        aria-expanded={open ? 'true' : undefined}
+      >
+        Workspace
+      </HelperPanelButton>
+      <Menu
+        open={open}
+        onClose={handleClose}
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+      >
+        <MenuItem
+          onClick={() => {
+            track({
+              action: 'Start Creating Workspace',
+              label: currentDataset?.hubmap_id,
+            });
+            setOpenCreateWorkspace(true);
+            handleClose();
+          }}
+        >
+          <ListItemIcon>
+            <WorkspacesIcon color="primary" fontSize="1.5rem" />
+          </ListItemIcon>
+          Launch New Workspace
+        </MenuItem>
+        <MenuItem
+          onClick={() => {
+            track({
+              action: 'Start Adding Dataset to Existing Workspace',
+              label: currentDataset?.hubmap_id,
+            });
+            openEditWorkspaceDialog();
+            handleClose();
+          }}
```

Comment:
Leverage `useCallback` for these onclicks? I feel like we've been consistent with that recently.